### ISSUE TITLE
Add ability to open codeplugs from manufacturer CPS

### ISSUE THREAD

**Issue by obi134** [hmatuschek/qdmr#64]
It would be great to open/import an existing codeplug, which was created by the manufacturer CPS. This would help to switch to this software.

**Comment by DC7IA** [comment]
It would be great, but you can also write it to the radio using the manufacturer's CPS and then read it. :D

**Comment by obi134** [comment]
Sure, but is there any difference in the data? Is the format of saved CP from Anytone different to the one which can be read back from device?

**Comment by hmatuschek** [comment]
Yes, there will be differences. I currently plan to extend the text format to allow for device specific settings as a *power user feature*. This would then be fully equivalent to the manufacturer code-plug. However, this is a huge undertaking. Just the parser and sanity check of all settings of each radio will be a large task. So I would rather postpone this feature and focus on stability and UI improvements.

**Comment by hmatuschek** [comment]
Concerning the import of manufacturer code plugs: For some radios it would be OK, as the binary manufacturer code plug is basically a memory image of what is written into the device. This is certainly true for the GD-77, RD-5R and to some extend also for the MD-UV390. For other manufacturer, especially AnyTone, the binary code plug has simply nothing to do with the data written into the device. In these cases I basically have to double the implementation work. 

However, I understand that it would be a nice feature. So I keep this bug request open but do not expect to see this feature soon.

**Comment by wp4vw** [comment]
I believe it would be great be able to use the DCF created by several OEM CPS’s into qDMR. I truly love this software and more that is one of the few that works on my intel based macs.
